app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialItemsViewHolder.java

```diff
@@ -153,4 +173,117 @@ private void manageMediaVisibility(EditorialContent editorialItem) {
       }
     }
   }
+
+  private void setPlaceHolderInfo(String appName, String image, float rating) {
+    ImageLoader.with(itemView.getContext())
+        .load(image, appCardImage);
+    appCardImage.setVisibility(View.VISIBLE);
+    if (rating == 0) {
+      appCardRating.setText(R.string.appcardview_title_no_stars);
+    } else {
+      appCardRating.setText(oneDecimalFormat.format(rating));
+    }
+    appCardRatingLayout.setVisibility(View.VISIBLE);
+    appCardNameWithRating.setText(appName);
+    appCardNameWithRating.setVisibility(View.VISIBLE);
+    appCardLayout.setVisibility(View.VISIBLE);
+  }
+
+  public boolean isVisible(float screenHeight, float screenWidth) {
+    final Rect placeHolderPosition = new Rect();
+    appCardLayout.getLocalVisibleRect(placeHolderPosition);
+    final Rect screen =
+        new Rect(0, 0, (int) screenWidth, (int) screenHeight - appCardLayout.getHeight() * 2);
+    return placeHolderPosition.intersect(screen);
+  }
+
+  public View getPlaceHolder() {
+    return appCardLayout;
+  }
+
+  public void setPlaceHolderDownloadingInfo(DownloadModel downloadModel) {
+    downloadInfoLayout.setVisibility(View.VISIBLE);
+    cardInfoLayout.setVisibility(View.GONE);
+    setDownloadState(downloadModel.getProgress(), downloadModel.getDownloadState());
+  }
+
+  public void setPlaceHolderDefaultStateInfo(DownloadModel downloadModel, String update,
+      String install, String open) {
+    downloadInfoLayout.setVisibility(View.GONE);
+    cardInfoLayout.setVisibility(View.VISIBLE);
+    setButtonText(downloadModel, update, install, open);
+  }
+
+  private void setButtonText(DownloadModel model, String update, String install, String open) {
+    DownloadModel.Action action = model.getAction();
+    switch (action) {
+      case UPDATE:
+        appCardButton.setText(update);
+        break;
+      case INSTALL:
+        appCardButton.setText(install);
+        break;
+      case OPEN:
+        appCardButton.setText(open);
+        break;
+    }
+  }
+
+  private void setDownloadState(int progress, DownloadModel.DownloadState downloadState) {
+    LinearLayout.LayoutParams pauseShowing =
+        new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT, 4f);
+    LinearLayout.LayoutParams pauseHidden =
+        new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT, 2f);
+    switch (downloadState) {
+      case ACTIVE:
```

Comment:
Each one of these cases could be extract to a method to improve code readibility.

Reply:
I think the suggestion doesn’t add much readibility